modules/product/src/Plugin/Field/FieldWidget/ProductVariationAttributesWidget.php

```diff
@@ -232,10 +232,11 @@ protected function getAttributeInfo(ProductVariationInterface $selected_variatio
     $field_definitions = $this->attributeFieldManager->getFieldDefinitions($selected_variation->bundle());
     $field_map = $this->attributeFieldManager->getFieldMap($selected_variation->bundle());
     $field_names = array_column($field_map, 'field_name');
+    $attribute_names = array_column($field_map, 'attribute_id');
     $index = 0;
     foreach ($field_names as $field_name) {
       /** @var \Drupal\commerce_product\Entity\ProductAttributeInterface $attribute_type */
-      $attribute_type = $this->attributeStorage->load(substr($field_name, 10));
+      $attribute_type = $this->attributeStorage->load($attribute_names[$index]);
```

Comment:
I agree substr() here is a weird thing but could it be better to have this instead:

`$attribute_type = $this->attributeStorage->load($field_map[$index]['attribute_id']);`

That way we have no additional lines and variables introduced.